src/util/search.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * SelectMultipleFilter needs to parse values from format
+ * "has_all:a,b,c,d" or "a,b,c,d"
+ */
+export const parseSelectFilterOptions = uriComponentValue => {
+  return uriComponentValue && uriComponentValue.indexOf(':') >= 0
+    ? uriComponentValue.split(':')[1].split(',')
+    : uriComponentValue && uriComponentValue.indexOf(':') < 0
+    ? uriComponentValue.split(',')
+    : [];
+};
+
+const flatten = (acc, val) => acc.concat(val);
+
+/**
+ * Check if the filter is currently active.
+ */
+export const findActiveFilter = (filterIds, urlQueryParams, filterConfigs) => {
+  const activeQueryParamKeys = filterConfigs
+    .filter(config => filterIds.includes(config.id))
+    .map(config => config.queryParamNames)
+    .reduce(flatten, []);
```

Comment:
A matter of taste, but as JS array ops are not efficiently collected into a single pass, you could achieve all this with a single `reduce` instead of going through the array three times.

Reply:
Yeah, reduce should be more efficient in this setup. I'm just not sure that most of our clients are familiar with `reduce`. So, lately, I have started to avoid `reduce` in places where it's more clear to have readable logic: filter > map > etc. 

Anyway, I change this to reduce format - especially, since I had to swap `flat()` call to `reduce` because Netlify/Node didn't support it.